lms/djangoapps/courseware/courses.py

```diff
@@ -141,6 +151,11 @@ def check_course_access(course, user, action, check_if_enrolled=False):
         if not CourseEnrollment.is_enrolled(user, course.id):
             raise CourseAccessRedirect(reverse('about_course', args=[unicode(course.id)]))
 
+    # Redirect if the user must answer a survey before entering the course.
+    if check_survey_complete:
```

Comment:
Especially since the default is True, do we need to check that `action` is `load`?  I still feel like I wish this could be in has_access.  We all need wishes. :)